### PR TITLE
DOC: instrument descriptions

### PR DESCRIPTION
**Pull request by jklenzing**
Going to do this in several smaller review friendly pulls.
- Fixes formatting and broken links in supported_instruments.rst
- Updates top level instrument docstrings for better support in Sphinx

tested via `make html`

### CODE REVIEW

**Comment by jklenzing** [comment]
Old Style:
Screen Shot 2020-07-23 at 8 55 43 PM

New Style:
Screen Shot 2020-07-23 at 10 31 03 PM

File: pysat/instruments/templates/template_cdaweb_instrument.py [hunk 1]
```diff
@@ -9,29 +9,37 @@
...
 Note
 ----
-::
+- Add any notes here
 
-    Notes
 
 Warnings
 --------
+- Add any warnings here
```
Comment (aburrell) [inline pysat/instruments/templates/template_cdaweb_instrument.py:30]: Recommend removing optional categories from template.  But there is good reason to keep them, so maybe we should discuss this.

File: pysat/instruments/timed_saber.py [hunk 2]
```diff
@@ -8,44 +8,42 @@
```
Comment (aburrell) [inline pysat/instruments/timed_saber.py:8]: Is this the best place for the "rules of the road"?

File: pysat/instruments/ucar_tiegcm.py [hunk 3]
```diff
@@ -4,21 +4,20 @@
...
-sat_id : string
+sat_id
     None supported
 
+
 Note
 ----
-::
-
     Loads into xarray format.
```
Comment (aburrell) [inline pysat/instruments/ucar_tiegcm.py:21]: Is this necessary?

File: pysat/instruments/templates/template_cdaweb_instrument.py (hunk 1 above)
Comment (jklenzing) [reply pysat/instruments/templates/template_cdaweb_instrument.py:30]: I was thinking of showing how this should be formatted in the templates to help move toward a uniform doc style.  One of the style errors I noticed is that if there are not two blank lines after the warnings, the next line (usually author information) gets added to the warnings box in the html.  What if there was a note that these are optional and should be removed if unused?

File: pysat/instruments/ucar_tiegcm.py (hunk 3 above)
Comment (jklenzing) [reply pysat/instruments/ucar_tiegcm.py:21]: Probably not, now that we have multiple xarray instruments

File: pysat/instruments/timed_saber.py (hunk 2 above)
Comment (jklenzing) [reply pysat/instruments/timed_saber.py:8]: How about as a note?

**Comment by aburrell** [comment]
In rocsat_ivm, recommend removing "Returns" from the "clean" function docstring and replace the "return None"  within that function with "return".  This may fix the coveralls issues in the related pulls and also clean up the docstring.

File: pysat/instruments/templates/template_cdaweb_instrument.py (hunk 1 above)
Comment (aburrell) [reply pysat/instruments/templates/template_cdaweb_instrument.py:30]: Sounds good.  Also add that note about the two blank lines.

**Comment by jklenzing** [comment]
Since this pull is pretty big, I was planning on fixing the function level docstrings in a separate pull, including:
- parens in parameter lists
- removal of "returns void" lines

Also, since the ICON instruments have changed significantly, I plan to fix the docstrings for those as part of that pull.

File: pysat/instruments/timed_saber.py (hunk 2 above)
Comment (aburrell) [reply pysat/instruments/timed_saber.py:8]: Sure, or maybe reference (not positive that's right either)?  Is this info included in the ackn or ref functions?

Comment (jklenzing) [reply pysat/instruments/timed_saber.py:8]: ackn and ref are still mostly unused throughout the instruments.  Been working on those as part of the 3.0 branch.

**Pull request merged**